eclass/zig-toolchain.eclass

```diff
@@ -0,0 +1,382 @@
+# @PRE_INHERIT
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# If set to a non-empty value, all logic in zig-toolchain and
+# zig-build eclasses will be considered optional.  No dependencies
+# will be added and no phase functions will be exported.
+#
+# For zig-toolchain.eclass users:
+# You have to add Zig dependency in your BDEPEND manually and call
+# at least "zig-toolchain_setup" before using "ezig".
+#
+# For zig-build.eclass users: see documentation in zig-build.eclass
+# instead.
+if [[ ! ${ZIG_OPTIONAL} ]]; then
+	BDEPEND="|| (
+		dev-lang/zig:${ZIG_SLOT}
+		dev-lang/zig-bin:${ZIG_SLOT}
+	)"
+fi
+
+# @ECLASS_VARIABLE: ZIG_TARGET
+# @DESCRIPTION:
+# Zig target triple to use.  Has the following format:
+# arch-os[.os_version_range]-abi[.abi_version]
+# Can be passed as:
+# * "-target " option in "zig test" or "zig build-exe/lib/obj",
+# * "-Dtarget=" option in "zig build"
+#   (if project uses "std.Build.standardTargetOptions").
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+#
+# Example:
+# @CODE
+# # Autodetected by Zig:
+# native
+# # Machine running Linux x86_64 system, with glibc:
+# x86_64-linux-gnu
+# # Similar to above, but versions are passed explicitly:
+# x86_64-linux.6.1.12...6.6.16-gnu.2.38
+# # Machine running Linux PPC64 little-endian system, with musl
+# powerpc64le-linux-musl
+# @CODE
+#
+
+# @ECLASS_VARIABLE: ZIG_CPU
+# @DESCRIPTION:
+# Zig target CPU and features to use.  Has the following format:
+# family_name(\+enable_feature|\-disable_feature)*
+# Can be passed as:
+# * "-mcpu " option in "zig test" or "zig build-exe/lib/obj",
+# * "-Dcpu=" option in "zig build"
+#   (if project uses "std.Build.standardTargetOptions").
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+#
+# Example:
+# @CODE
+# # Autodetected by Zig:
+# native
+# # AMD Zen 2 processor
+# znver2
+# # x86_64 processor, X87 support enabled, SSE2 support disabled
+# x86_64+x87-sse2
+# @CODE
+
+# @ECLASS_VARIABLE: ZIG_EXE
+# @DESCRIPTION:
+# Absolute path to the used Zig executable.
+#
+# Please note that when passing one flag several times with different
+# values:
+# * (only "zig build") in "-Dbar=false -Dbar" form:
+#   errors due to conflict of flags,
+# * (only "zig build") in "-Dbar=false -Dbar=true" form:
+#   "bar" becomes a list, which is likely not what you want,
+# * in "-fbar -fno-bar" form:
+#   latest value overwrites values before.
+# Example above shows only boolean option, but it is same with other
+# types of options (enums, "std.zig.BuildId", "std.SemanticVersion",
+# integers, strings, etc.).
+#
+# Can be overriden by user.  If not overriden, then set by
+# "zig-toolchain_setup".
+
+# @ECLASS_VARIABLE: ZIG_VER
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Zig version found during "zig-toolchain_find_installation",
+# as reported in dev-lang/zig-${PV} PV part.
+#
+# Example:
+# @CODE
+# 0.13.0
+# @CODE
+
+# @FUNCTION: zig-toolchain_get_target
```

Comment:
The name doesn't suggest "translating" at all. `*triple_to_zig_target` or something like that would probably work better.

Reply:
I copied style of rust-toolchain.eclass here, where similar function was named `rust_abi` and it also "translates" CHOST (or 1 parameter) value. Should I name it like previously in pre-rewrite Zig ebuilds, like `c_triple_to_zig_target`? (BTW in the future it might become quadriple for Zig, that's why I'm not writing `zig_triple` here)

Reply:
`zig-utils_c_tuple_to_zig_target`?

Reply:
WFM. I suppose you can even skip `tuple`, it's a "target" in C as well.

Reply:
966f169f3fb2...a9b50921a624